OpenSim/Region/Framework/Scenes/SceneObjectPartInventory.cs

```diff
@@ -695,8 +695,16 @@ private void _ReplaceItem(SceneObjectPart part, UUID itemID, bool allowedDrop, b
         {
             TaskInventoryItem newItem = null;
             bool scriptUpdate = false;
-            int type = m_items[itemID].InvType;
-            if (type == 10) // Script
+
+            // Avoid a race condition where the viewer might send what results in a second _Replace(itemID, null) 
+            // for a delete which is a *duplicate*. This can happen if the user deletes the TaskInventoryItem 
+            // a *second* time before the viewer updates the Contents to remove it. This change just handle 
+            // the itemID no longer being present in the items list.
+            if (!m_items.ContainsKey(itemID))
```

Comment:
This still leaves the potential for the crash to happen on the following InventoryType type = (InventoryType) m_items[itemID].InvType call. I suggest instead using if (m_items.TryGetValue()) and only proceeding if the value exists